chalicelib/app_utils.py

```diff
@@ -230,8 +236,35 @@ def query_params_to_literals(params):
     return params
 
 
-def trim_output(output, max_size=100000):
+def get_size(obj, seen=None):
+    """ Recursively finds size of objects
+        Taken directly from: https://goshippo.com/blog/measure-real-size-any-python-object/
     """
+    size = sys.getsizeof(obj)
+    if seen is None:
+        seen = set()
+    obj_id = id(obj)
+    if obj_id in seen:
+        return 0
+    # Important mark as seen *before* entering recursion to gracefully handle
+    # self-referential objects
+    seen.add(obj_id)
+    if isinstance(obj, dict):
+        size += sum([get_size(v, seen) for v in obj.values()])
+        size += sum([get_size(k, seen) for k in obj.keys()])
+    elif hasattr(obj, '__dict__'):
+        size += get_size(obj.__dict__, seen)
+    elif hasattr(obj, '__iter__') and not isinstance(obj, (str, bytes, bytearray)):
```

Comment:
Yeah, this is a tricky piece of code. The article that came with it may help answer your questions. I took it because it performs better then the old code that relies on `json.dumps` in `trim_output`.